torch/nn/modules/container.py

```diff
@@ -276,3 +392,126 @@ def extra_repr(self):
+        del self._parameters[key]
+
+    def __len__(self):
+        return len(self._parameters)
+
+    def __iter__(self):
+        return iter(self._parameters.keys())
+
+    def __contains__(self, key):
+        return key in self._parameters
+
+    def clear(self):
+        """Remove all items from the ParameterDict.
+        """
+        self._parameters.clear()
+
+    def pop(self, key):
+        r"""Remove key from the ParameterDict and return its parameter.
+
+        Arguments:
+            key (string): key to pop from the ParameterDict
+        """
+        v = self[key]
+        del self[key]
+        return v
+
+    def keys(self):
+        r"""Return an iterable of the ParameterDict’s keys.
+        """
+        return self._parameters.keys()
+
+    def items(self):
+        r"""Return an iterable of the ParameterDict’s key/value pairs.
+        """
+        return self._parameters.items()
+
+    def values(self):
+        r"""Return an iterable of the ParameterDict’s values.
+        """
+        return self._parameters.values()
+
+    def update(self, parameters):
+        r"""Update the ParameterDict with the key/value pairs from a mapping or
+        an iterable, overwriting existing keys.
+
+        Arguments:
+            parameters (iterable): a mapping (dictionary) of
+                (string : :class:`~torch.nn.Parameter``) or an iterable of
+                key/value pairs of type (string, :class:`~torch.nn.Parameter``)
+        """
+        if not isinstance(parameters, Iterable):
+            raise TypeError("ParametersDict.update should be called with an "
+                            "iterable of key/value pairs, but got " +
+                            type(parameters).__name__)
+
+        if isinstance(parameters, Mapping):
+            for key, parameter in parameters.items():
+                self[key] = parameter
+        else:
+            for j, p in enumerate(parameters):
+                if not isinstance(p, Iterable):
+                    raise TypeError("ParameterDict update sequence element "
+                                    "#" + str(j) + " should be Iterable; is" +
+                                    type(p).__name__)
+                if not len(p) == 2:
+                    raise ValueError("ParameterDict update sequence element "
+                                     "#" + str(j) + " has length " + str(len(p)) +
+                                     "; 2 is required")
+                self[p[0]] = p[1]
+
+    def extra_repr(self):
+        tmpstr = ''
+        for k, p in self._parameters.items():
+            size_str = 'x'.join(str(size) for size in p.size())
+            device_str = '' if not p.is_cuda else ' (GPU {})'.format(p.get_device())
+            parastr = 'Parameter containing: [{} of size {}{}]'.format(
+                torch.typename(p.data), size_str, device_str)
+            tmpstr = tmpstr + '  (' + str(k) + '): ' + parastr + '\n'
+        return tmpstr
```

Comment:
Can you post an example of this?

Reply:
ModuleDict(
  (choice1): Sigmoid()
  (choice2): ReLU()
)

Reply:
Adapted from ParameterList
```
ParameterDict(
    (choice1): Parameter containing: [torch.FloatTensor of size 10x20]
    (choice2): Parameter containing: [torch.FloatTensor of size 10x30]
  
)
```

Reply:
This printing format is probably fine for the purpose of this PR. But we should also think about if we want to update it and ParameterDict to be more consistent with the current print format. 

The extra `\n` is ugly though. Can you add to a list, then do `'\n'.join(list_or_strs)` instead?

Reply:
opened an issue for ParameterList at https://github.com/pytorch/pytorch/issues/8900

Reply:
> The extra \n is ugly though. Can you add to a list, then do '\n'.join(list_or_strs) instead?

I will PR this with ParameterList, ModuleList, ModuleDict, Sequential in the above issue